payment/internal/server/error.go

```diff
@@ -0,0 +1,21 @@
+package server
+
+import (
+	"github.com/omise/omise-go"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc/codes"
+)
+
+func InferCodeFromOmiseError(err error) codes.Code {
+	omiseError := &omise.Error{}
+	switch errors.As(err, &omiseError); omiseError.StatusCode {
+	case 400:
+		return codes.InvalidArgument
+	case 403:
+		return codes.FailedPrecondition
```

Comment:
I'm not sure what 403 means for Omise API, but I'd think it's probably `codes.PermissionDenied` no? Just ignore this if im wrong.

Reply:
403 means payment service using expired key. I think it should be internal.